cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -555,4 +608,37 @@ public void close() {
       baseScanner.close();
     }
   }
+
+  // scanner that matches column values based upon exact match
+  private class IndexScanner extends AbstractIndexScanner {
+    private final byte[] value;
+
+    public IndexScanner(Scanner baseScanner, byte[] column, byte[] value) {
+      super(baseScanner, column);
+      this.value = value;
+    }
+
+    @Override
+    protected boolean matches(byte[] columnValue) {
+      return Bytes.equals(columnValue, value);
+    }
+  }
+
+  // scanner that matches column values based upon range
+  private class IndexRangeScanner extends AbstractIndexScanner {
+    private final byte[] start;
+    private final byte[] end;
+
+    public IndexRangeScanner(Scanner baseScanner, byte[] column, byte[] start, byte[] end) {
+      super(baseScanner, column);
+      this.start = start;
+      this.end = end;
+    }
+
+    @Override
```

Comment:
I am wondering whether this check is always true. Since the scan is from <column>0<start> to <column>0<end>, and the extracted column value is always a prefix of what follows <column>0, can it ever happen that it is not in the range?